packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1613,123 +1712,201 @@ class _TimePickerDialogState extends State<_TimePickerDialog> {
   }
 
   void _handleOk() {
+    if (_entryMode == TimePickerEntryMode.input) {
+      final FormState form = _formKey.currentState;
+      if (!form.validate()) {
+        setState(() => _autoValidate = true);
+        return;
+      }
+      form.save();
+    }
     Navigator.pop(context, _selectedTime);
   }
 
+  Size _dialogSize(BuildContext context) {
+    final Orientation orientation = MediaQuery.of(context).orientation;
+    final ThemeData theme = Theme.of(context);
+    // Constrain the textScaleFactor to prevent layout issues.
+    final double textScaleFactor = math.min(MediaQuery.of(context).textScaleFactor, 1.1);
+
+    double timePickerWidth;
+    double timePickerHeight;
+    switch (_entryMode) {
+      case TimePickerEntryMode.dial:
+        switch (orientation) {
+          case Orientation.portrait:
+            timePickerWidth = _kTimePickerWidthPortrait;
+            timePickerHeight = theme.materialTapTargetSize == MaterialTapTargetSize.padded
+                ? _kTimePickerHeightPortrait
+                : _kTimePickerHeightPortraitCollapsed;
+            break;
+          case Orientation.landscape:
+            timePickerWidth = _kTimePickerWidthLandscape * textScaleFactor;
+            timePickerHeight = theme.materialTapTargetSize == MaterialTapTargetSize.padded
+                ? _kTimePickerHeightLandscape
+                : _kTimePickerHeightLandscapeCollapsed;
+            break;
+        }
+        break;
+      case TimePickerEntryMode.input:
+        timePickerWidth = _kTimePickerWidthPortrait;
+        timePickerHeight = _kTimePickerHeightInput;
+        break;
+    }
+    return Size(timePickerWidth, timePickerHeight * textScaleFactor);
+  }
+
   @override
   Widget build(BuildContext context) {
     assert(debugCheckHasMediaQuery(context));
     final MediaQueryData media = MediaQuery.of(context);
     final TimeOfDayFormat timeOfDayFormat = localizations.timeOfDayFormat(alwaysUse24HourFormat: media.alwaysUse24HourFormat);
     final bool use24HourDials = hourFormat(of: timeOfDayFormat) != HourFormat.h;
     final ThemeData theme = Theme.of(context);
+    final ShapeBorder shape = TimePickerTheme.of(context).shape ?? _kDefaultShape;
+    final Orientation orientation = MediaQuery.of(context).orientation;
 
-    final Widget picker = Padding(
-      padding: const EdgeInsets.all(16.0),
-      child: AspectRatio(
-        aspectRatio: 1.0,
-        child: _Dial(
-          mode: _mode,
-          use24HourDials: use24HourDials,
-          selectedTime: _selectedTime,
-          onChanged: _handleTimeChanged,
-          onHourSelected: _handleHourSelected,
-        ),
-      ),
-    );
-
-    final Widget actions = ButtonBar(
+    Color toggleColor;
+    switch (theme.brightness) {
```

Comment:
Shouldn't we be using ColorScheme colors here?

Reply:
Done